### PR TITLE
aspeed: crypto: fix digest incorrect problem

### PR DESCRIPTION
**Pull request by Neal-liu**
These patch series include 2 parts:
1. Fix processing sequence wrong and typos
2. Lock HACE engine into Secure World

### CODE REVIEW

**Comment by Neal-liu** [comment]
We came out better solution to fix the issue, I'll update this PR ASAP.
Sorry for your inconvenience If you are reviewing this PR.

File: core/drivers/crypto/aspeed/hace_ast2600.c [hunk 1]
```diff
@@ -216,11 +221,10 @@ static TEE_Result ast2600_hace_final(struct crypto_hash_ctx *ctx,
 	uint64_t dbits[2] = { };
 	uint64_t dbits_be[2] = { };
 	struct ast2600_hace_ctx *hctx = NULL;
+	size_t length;
```
Comment (jforissier) [inline core/drivers/crypto/aspeed/hace_ast2600.c:224]: `= 0;`

File: core/drivers/crypto/aspeed/hace_ast2600.c [hunk 2]
```diff
@@ -216,11 +221,10 @@ static TEE_Result ast2600_hace_final(struct crypto_hash_ctx *ctx,
...
 
 	hctx = container_of(ctx, struct ast2600_hace_ctx, hash_ctx);
-
-	if (len < hctx->dgt_size)
-		return TEE_ERROR_BAD_PARAMETERS;
+	length = len < hctx->dgt_size ? len : hctx->dgt_size;
```
Comment (jforissier) [inline core/drivers/crypto/aspeed/hace_ast2600.c:227]: `MIN(len, hctx->dgt_size)`

File: core/arch/arm/plat-aspeed/platform_ast2600.c [hunk 3]
```diff
@@ -85,14 +87,20 @@ void console_init(void)
 void plat_primary_init_early(void)
 {
 	vaddr_t ahbc_virt = 0;
+	uint32_t tzm_perm;
```
Comment (jforissier) [inline core/arch/arm/plat-aspeed/platform_ast2600.c:90]: `= 0;`

File: core/drivers/crypto/aspeed/hace_ast2600.h [hunk 4]
```diff
@@ -15,3 +15,4 @@ static inline TEE_Result ast2600_drvcrypt_register_hash(void)
 	return TEE_ERROR_NOT_SUPPORTED;
 }
 #endif
+#endif
```
Comment (jforissier) [inline core/drivers/crypto/aspeed/hace_ast2600.h:18]: This change should be in a separate commit. I suppose it fixes a build issue.

**Review by jforissier — COMMENTED** [review]
Hi @Neal-liu,

Comments below. With comments addressed:

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by Neal-liu** [comment]
> Hi @Neal-liu,
> 
> Comments below. With comments addressed:
> 
> `Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

Tag applied, thanks for your review.
